src/lib/components/EventQAndA/index.js

```diff
@@ -0,0 +1,80 @@
+import {BaseElement} from '../BaseElement';
+import './_styles.scss';
+
+/**
+ * Filters out Q&A's based on `data-category` property.
+ *
+ * @extends {BaseElement}
+ * @final
+ */
+class EventQAndA extends BaseElement {
+  constructor() {
+    super();
+    this.selectCategory = this.selectCategory.bind(this);
+
+    this.categories = new Set();
+    this.categorySelectElement = document.createElement('select');
+    this.categorySelectElement.classList.add('w-select--borderless');
+    this.categorySelectElement.addEventListener('change', this.selectCategory);
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    this.addDropDown();
+
+    this.querySelectorAll('[data-category]').forEach((element) => {
```

Comment:
I don't want to bikeshed but I think doing something like:

```js
  this.addEventListener('click', this.closeDetail);

// later
  closeDetail(e) {
    const category = e.target.closest('[data-category]');
    if (!category) {
      return;
    }
    this.categoryElements.forEach((cand) => {
      if (cand !== this) { ... }
    });
  }
```

is a bunch neater as we don't have to try to maintain a bunch of click listeners.

Reply:
Because of the padding on the elements I took this in a different direction, but have also moved away from all the click listeners (kinda), so it should be much neater.